Doc/c-api/typeobj.rst

```diff
@@ -1034,7 +1034,9 @@ and :c:data:`PyType_Type` effectively act as defaults.)
       the type, and the type object is INCREF'ed when a new instance is created, and
       DECREF'ed when an instance is destroyed (this does not apply to instances of
       subtypes; only the type referenced by the instance's ob_type gets INCREF'ed or
-      DECREF'ed).
+      DECREF'ed). Heap types must also support garbage collection as
+      they can form a `reference cycle <https://docs.python.org/3/c-api/gcsupport.html#supporting-cyclic-garbage-collection>`_
```

Comment:
I was trying to way to add this link using elements/references but I couldn't figure it out. Maybe there's a nicer way to do this, open to suggestions!

Reply:
I think the wording is good but I would move the link to the words "support garbage collection". Also, you should be able to use a :ref: link, like:
```
:ref:`support garbage collection <supporting-cycle-detection>`
```
(Using the link target that points to the top of `gcsupport.rst`.)

Reply:
Wow, I swear I tried that text for the reference link 😅 . I was also torn on where to place the link but I agree, it makes more sense where you suggested.